packages/walletconnect-v2/src/index.ts

```diff
@@ -86,27 +111,47 @@ export class WalletConnect extends Connector {
     this.events.emit(URI_AVAILABLE, uri)
   }
 
+  private async initializeProvider(
+    desiredChainId: number | undefined = this.defaultChainId
+  ): Promise<WalletConnectProvider> {
+    const ethProviderModule = await import('@walletconnect/ethereum-provider')
+    const rpcMap = this.rpcMap ? await getBestUrlMap(this.rpcMap, this.timeout) : undefined
+    const chainProps = this.getChainProps(this.chains, this.optionalChains, desiredChainId)
+
+    this.provider = await ethProviderModule.default.init({
+      ...this.options,
+      ...chainProps,
+      rpcMap,
+    })
+
+    return this.handleProviderEvents(this.provider)
+  }
+
+  private getChainProps(
+    chains: number[] | ArrayOneOrMore<number> | undefined,
+    optionalChains: number[] | ArrayOneOrMore<number> | undefined,
+    desiredChainId: number | undefined = this.defaultChainId
+  ): ChainsProps {
+    // Reorder chains and optionalChains if necessary
+    const orderedChains = getChainsWithDefault(chains, desiredChainId)
+    const orderedOptionalChains = getChainsWithDefault(optionalChains, desiredChainId)
+
+    // Validate and return the result
+    if (isArrayOneOrMore(orderedChains)) {
+      return { chains: orderedChains, optionalChains: orderedOptionalChains }
+    } else if (isArrayOneOrMore(orderedOptionalChains)) {
+      return { optionalChains: orderedOptionalChains, chains: orderedChains }
```

Comment:
return the natural ordering
```suggestion
      return { chains: orderedChains, optionalChains: orderedOptionalChains }
```